include/vkwrap/vkwrap/pipeline_cfgs.h

```diff
@@ -0,0 +1,352 @@
+};
+
+template <typename Derived> class RasterizerCfg
+{
+  public:
+    // clang-format off
+    void make( vk::GraphicsPipelineCreateInfo& pipeline_create_info )
+    {
+        m_rasterizer.setDepthClampEnable( VK_FALSE );
+        m_rasterizer.setRasterizerDiscardEnable( VK_FALSE );
+        m_rasterizer.setPolygonMode( vk::PolygonMode::eFill );
+        m_rasterizer.setLineWidth( 1.0f );
+        m_rasterizer.setCullMode( vk::CullModeFlagBits::eBack );
+        m_rasterizer.setFrontFace( vk::FrontFace::eClockwise );
+        m_rasterizer.setDepthBiasEnable( VK_FALSE );
+        m_rasterizer.setDepthBiasConstantFactor( 0.0f );
+        m_rasterizer.setDepthBiasClamp( 0.0f );
+        m_rasterizer.setDepthBiasSlopeFactor( 0.0f );
+
+	pipeline_create_info.setPRasterizationState( &m_rasterizer );
+    }
+    // clang-format on
+
+  protected:
+    RasterizerCfg() = default;
+    RasterizerCfg( RasterizerCfg&& ) = default;
+    RasterizerCfg( const RasterizerCfg& ) = default;
+    RasterizerCfg& operator=( const RasterizerCfg& ) = default;
+    RasterizerCfg& operator=( RasterizerCfg&& ) = default;
+    ~RasterizerCfg() = default;
+
+  private:
+    vk::PipelineRasterizationStateCreateInfo m_rasterizer;
+};
+
+template <typename Derived> class InputAssemblyCfg
+{
+  public:
+    // clang-format off
+    void make( vk::GraphicsPipelineCreateInfo& pipeline_create_info )
+    {
+        m_input_assembly.setTopology( vk::PrimitiveTopology::eTriangleList );
+        m_input_assembly.setPrimitiveRestartEnable( VK_FALSE );
+
+	pipeline_create_info.setPInputAssemblyState( &m_input_assembly );
+    }
+
+    // clang-format on
+
+  protected:
+    InputAssemblyCfg() = default;
+    InputAssemblyCfg( InputAssemblyCfg&& ) = default;
+    InputAssemblyCfg( const InputAssemblyCfg& ) = default;
+    InputAssemblyCfg& operator=( const InputAssemblyCfg& ) = default;
+    InputAssemblyCfg& operator=( InputAssemblyCfg&& ) = default;
+    ~InputAssemblyCfg() = default;
+
+  private:
+    vk::PipelineInputAssemblyStateCreateInfo m_input_assembly;
+};
+
+template <typename Derived> class PipelineLayoutCfg
+{
+  public:
+    template <typename Layout = ranges::empty_view<vk::UniqueDescriptorSetLayout>>
+    Derived& withPipelineLayout( vk::Device device, Layout&& layouts = {} ) &
+    {
+        m_layouts = ranges::views::transform( layouts, []( auto&& elem ) { return elem.get(); } ) | ranges::to_vector;
+
+        vk::PipelineLayoutCreateInfo layout_create_info{
+            .setLayoutCount = static_cast<uint32_t>( m_layouts.size() ),
+            .pSetLayouts = m_layouts.data() };
+
+        m_pipeline_layout = device.createPipelineLayoutUnique( layout_create_info );
+
+        return static_cast<Derived&>( *this );
+    }
+
+    // clang-format off
+    void make( vk::GraphicsPipelineCreateInfo& pipeline_create_info )
+    {
+	pipeline_create_info.setLayout( m_pipeline_layout.get() );
+    }
+
+    // clang-format on
```

Comment:
Formatting is broken. If you intended to force clang-format to keep the function multiline, then just add a trailing comment: ```....m_pipeline_layout.get() ); -> // <- ```

Reply:
Fixed